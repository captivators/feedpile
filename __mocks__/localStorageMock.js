const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"faiz"}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
