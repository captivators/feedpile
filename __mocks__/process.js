const process = {
  "env" : {
    "AUTH_ID":"test",
    "AUTH_DOMAIN":"user_id"
  }
}
global.process = process;
