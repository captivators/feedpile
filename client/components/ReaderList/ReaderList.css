<<<<<<< HEAD
.reader-list-container {
  max-width: 600px;
  padding-top: 30px;
  margin: auto;
}
=======
.container { /* Masonry container */
    column-count: 4;
    column-gap: 1em;
}

.item { /* Masonry bricks or child elements */
    background-color: #eee;
    display: inline-block;
    margin: 0 0 1em;
    width: 100%;
}
>>>>>>> added spec folder
