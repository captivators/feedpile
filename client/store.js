import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, compose (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
