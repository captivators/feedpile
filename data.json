[
  {
    "title":"Tillerson Warns Russia that Assad Era Is 'Coming to an End'",
    "publisher":"New York Times",
    "image":"https://static01.nyt.com/images/2017/04/12/world/12Diplo/12Diplo-moth.jpg",
    "url": "http://www.nytimes.com/2017/04/11/world/europe/russia-syria-rex-tillerson.html",
    "description":"The secretary of state said President Vladimir Putin of Russia must end his alliance with Bashar al-Assad or risk becoming irrelevant in the Middle East."
  },
  {
    "title":"On Trump's Syria Strategy, One Voice Is Missing: Trump's",
    "publisher":"New York Times",
    "image":"https://static01.nyt.com/images/2017/04/11/us/11TRUMPSYRIA-01/11Tillerson-moth.jpg",
    "url": "http://www.nytimes.com/2017/04/10/world/middleeast/syria-trump-assad.html",
    "description":"Administration officials have described starkly different foreign policies since last weekâ€™s cruise missile strike, and President Trump has yet to clarify."
  },
  {
    "title":"Feud Over Syria Missile Strike May Have an Upside for U.S. and Russia",
    "publisher":"New York Times",
    "image":"https://s15.postimg.org/6jredm53v/news.png",
    "url": "http://www.nytimes.com/2017/04/11/world/missile-strike-syria-reset-russia-us-relations.html",
    "description":"The American attack has freed leaders in Washington and Moscow from the constraints that come with the perception of a special, friendly relationship."
  },
  {
    "title":"Trump Says China Will Get Better Trade Deal if It Solves 'North Korean Problem'",
    "publisher":"New York Times",
    "image":"https://static01.nyt.com/images/2017/04/12/world/12TrumpAsia/12TrumpAsia-moth.jpg",
    "url": "http://www.nytimes.com/2017/04/11/world/asia/trump-china-trade-north-korea.html",
    "description":"President Trump's Twitter posts capture his quandary in dealing with the nuclear threat from North Korea: Only China can realistically force a change in the behavior of Kim Jong-un."
  },
  {
    "title":"Move of U.S. Warships Shows Trump Has Few Options on North Korea",
    "publisher":"New York Times",
    "image":"https://static01.nyt.com/images/2017/04/11/us/11Korea-01/11Korea1-moth.jpg",
    "url": "http://www.nytimes.com/2017/04/10/world/asia/north-korea-china-south.html",
    "description":"President Trump is confronting North Korean aggression with a set of tools that has had limited success for the United States in the past."
  },
  {
    "title":"The Latest Test for the White House? Pulling Off Its Easter Egg Roll",
    "publisher":"New York Times",
    "image":"https://static01.nyt.com/images/2017/04/05/us/05eggroll2/05eggroll2-moth.jpg",
    "url": "http://www.nytimes.com/2017/04/11/us/politics/white-house-easter-egg-roll-trump.html",
    "description":"With key staff positions unfilled and no on-site first lady, the White House is trying to organize the most elaborate and heavily scrutinized public event of the year."
  },
  {
    "title":"Trump Administration Hunts for Easter Eggs, and Senior Staff",
    "publisher":"New York Times",
    "image":"https://s15.postimg.org/6jredm53v/news.png",
    "url": "http://www.nytimes.com/video/us/politics/100000005018390/trump-administration-hunts-for-easter-eggs-and-senior-staff.html",
    "description":"The White House is as much as two months behind recent standards for presidential transitions, leaving 90 percent of the positions considered critical to leadership unfilled. It did, however, manage to order the eggs for the Easter egg roll."
  },
  {
    "title":"Secret A.T.F. Account Paid for $21,000 Nascar Suite and Las Vegas Trip",
    "publisher":"New York Times",
    "image":"https://static01.nyt.com/images/2017/04/11/us/11ATF/11ATF-moth.jpg",
    "url": "http://www.nytimes.com/2017/04/11/us/politics/alcohol-tobacco-firearms-atf.html",
    "description":"The revelations highlight the lax oversight at the A.T.F. that allowed agents and informants to spend millions while avoiding the normal accounting process."
  }
]
