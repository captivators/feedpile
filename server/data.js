module.exports = {
 "users":[
  {
    "username": "MightyMouse",
    "password": "938dsweoirwu",
    "articles": [
    { "article_id": 0, "readFlag": true },
    { "article_id": 1, "readFlag": false }
    ]
  },
  {
    "username": "codybot@netscape.biz",
    "password": "938dsweoirwu",
    "articles": [
    { "article_id": 0, "readFlag": true },
    { "article_id": 1, "readFlag": false },
    { "article_id": 2, "readFlag": false }
  ]
  }
],
"feeds": [
  {
    "url": "techcrunch.com/feed/atom/",
    "name": "TechCrunch"
  },
  {
    "url": "feed.slate.com/slate",
    "name": "Slate"
  },
  {
    "url": "http://rss.cnn.com/rss/cnn_topstories.rss",
    "name": "CNN"
  }
],
"articles": [
  {
    "article_url": "http://www.slate.com/articles/news_and_politics/jurisprudence/2017/04/how_the_supreme_court_will_change_after_the_nuclear_option.html",
    "date_written": "2017-04-08T22:16:01Z",
    "feedname": "Slate"
  },
  {
    "article_url": "https://techcrunch.com/2017/04/08/spotify-premium-director-robert-lamvik-leaves-company-for-meditation-app-headspace/",
    "date_written": "2017-04-08T16:13:43Z",
    "feedname": "TechCrunch"
  },
  {
    "article_url": "https://techcrunch.com/2017/04/08/spotify-premium-director-robert-lamvik-leaves-company-for-meditation-app-headspace/",
    "date_written": "2017-04-07T21:16:01Z",
    "feedname": "TechCrunch"
  },
 {
    "article_url": "https://techcrunch.com/2017/04/08/spotify-premium-director-robert-lamvik-leaves-company-for-meditation-app-headspace/",
    "date_written": "2017-04-07T21:16:01Z",
    "feedname": "TechCrunch"
 },
 {
  "article_url": "http://rss.cnn.com/~r/rss/cnn_topstories/~3/4PSdd4GKIao/index.html",
    "date_written": "2017-04-10T21:16:01Z",
    "feedname": "CNN"
 },
 {
  "article_url": "http://rss.cnn.com/~r/rss/cnn_topstories/~3/ycobN1EL5hE/believer-reza-aslan-who-are-aghori-sahdus-india-orig-ff.cnn",
    "date_written": "2017-03-13T21:16:01Z",
    "feedname": "CNN"
 },
 {
  "article_url": "http://rss.cnn.com/~r/rss/cnn_topstories/~3/GTyfDq9G4a4/index.html",
    "date_written": "2017-04-09T21:16:01Z",
    "feedname": "CNN"
 }
]

}