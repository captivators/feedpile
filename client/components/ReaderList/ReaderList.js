import React from 'react';
import ReaderListItem from '../ReaderListItem/ReaderListItem';
import {connect} from 'react-redux';
import { getArticlesFromDb } from '../../actions';
import './ReaderList.css';

class ReaderList extends React.Component {
  constructor (props){
    super(props);
  }
  componentDidMount() {
    if(!this.props.articles.length) this.props.getArticlesFromDb();
  }
  render() {
    return (
        <div className="reader-list-container">
          {this.props.articles.map((article, index) => (
              <ReaderListItem history={this.props.history}
                              article={article}
                              key={index}/>
          ))}
        </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    articles: state.articles
  }
};

export const Unwrapped = ReaderList;
export default connect(mapStateToProps, {getArticlesFromDb})(ReaderList);
