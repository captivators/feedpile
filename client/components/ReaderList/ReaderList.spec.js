import '../../../__mocks__/localStorageMock'
import React from 'react';
import {Provider} from 'react-redux';
import store from '../../store'
import ReaderList, {Unwrapped as UnwrappedReaderList} from './ReaderList';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import data from '../../../data.json'

test('ReaderList take a snapshot', () => {
  const component = shallow(<UnwrappedReaderList articles={data}/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
test('ReaderList should render a ReaderListItem for each article', () => {
  const component = render(<Provider store={store}><UnwrappedReaderList articles={data}/></Provider>);
  expect(component.find('.list-item-container').length).toEqual(data.length);
});
