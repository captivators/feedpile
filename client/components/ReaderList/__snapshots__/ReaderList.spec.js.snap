// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReaderList take a snapshot 1`] = `
<div
  className="reader-list-container"
>
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "The secretary of state said President Vladimir Putin of Russia must end his alliance with Bashar al-Assad or risk becoming irrelevant in the Middle East.",
        "image": "https://static01.nyt.com/images/2017/04/12/world/12Diplo/12Diplo-moth.jpg",
        "publisher": "New York Times",
        "title": "Tillerson Warns Russia that Assad Era Is 'Coming to an End'",
        "url": "http://www.nytimes.com/2017/04/11/world/europe/russia-syria-rex-tillerson.html",
      }
    }
    articleIndex={0}
  />
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "<img width=\\"680\\" height=\\"428\\" src=\\"https://tctechcrunch2011.files.wordpress.com/2017/04/execc81cution_de_marie_antoinette_le_16_octobre_1793-e1492125699842.jpg?w=680\\" class=\\"attachment-large size-large wp-post-image\\" alt=\\"\\" srcset=\\"https://tctechcrunch2011.files.wordpress.com/2017/04/execc81cution_de_marie_antoinette_le_16_octobre_1793-e1492125699842.jpg?w=680 680w, https://tctechcrunch2011.files.wordpress.com/2017/04/execc81cution_de_marie_antoinette_le_16_octobre_1793-e1492125699842.jpg?w=1360 1360w, https://tctechcrunch2011.files.wordpress.com/2017/04/execc81cution_de_marie_antoinette_le_16_octobre_1793-e1492125699842.jpg?w=150 150w, https://tctechcrunch2011.files.wordpress.com/2017/04/execc81cution_de_marie_antoinette_le_16_octobre_1793-e1492125699842.jpg?w=300 300w, https://tctechcrunch2011.files.wordpress.com/2017/04/execc81cution_de_marie_antoinette_le_16_octobre_1793-e1492125699842.jpg?w=768 768w\\" sizes=\\"(max-width: 680px) 100vw, 680px\\" />&nbsp;For every bad idea, there&#8217;s a crowdsourcing campaign with a soapbox and a big-ass megaphone, ready to make it even worse. Interested? In case you missed it, it&#8217;s not too late to vote to shake up the executive branch &#8212; surely the number one priority of every red-blooded American in these mostly ho-hum, sleepy times. As the giant stack of paper behind Office of Management&hellip; <a href=\\"https://techcrunch.com/2017/04/13/omb-reorganizing-executive-branch/?ncid=rss\\">Read More</a>",
        "image": "https://tctechcrunch2011.files.wordpress.com/2017/04/execc81cution_de_marie_antoinette_le_16_octobre_1793-e1492125699842.jpg?w=210&h=158&crop=1",
        "publisher": "New York Times",
        "title": "Vote now to eliminate your favorite government agency!",
        "url": "https://techcrunch.com/2017/04/13/omb-reorganizing-executive-branch/?ncid=rss",
      }
    }
    articleIndex={1}
  />
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "The American attack has freed leaders in Washington and Moscow from the constraints that come with the perception of a special, friendly relationship.",
        "image": "https://s15.postimg.org/6jredm53v/news.png",
        "publisher": "New York Times",
        "title": "Feud Over Syria Missile Strike May Have an Upside for U.S. and Russia",
        "url": "http://www.nytimes.com/2017/04/11/world/missile-strike-syria-reset-russia-us-relations.html",
      }
    }
    articleIndex={2}
  />
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "President Trump's Twitter posts capture his quandary in dealing with the nuclear threat from North Korea: Only China can realistically force a change in the behavior of Kim Jong-un.",
        "image": "https://static01.nyt.com/images/2017/04/12/world/12TrumpAsia/12TrumpAsia-moth.jpg",
        "publisher": "New York Times",
        "title": "Trump Says China Will Get Better Trade Deal if It Solves 'North Korean Problem'",
        "url": "http://www.nytimes.com/2017/04/11/world/asia/trump-china-trade-north-korea.html",
      }
    }
    articleIndex={3}
  />
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "President Trump is confronting North Korean aggression with a set of tools that has had limited success for the United States in the past.",
        "image": "https://static01.nyt.com/images/2017/04/11/us/11Korea-01/11Korea1-moth.jpg",
        "publisher": "New York Times",
        "title": "Move of U.S. Warships Shows Trump Has Few Options on North Korea",
        "url": "http://www.nytimes.com/2017/04/10/world/asia/north-korea-china-south.html",
      }
    }
    articleIndex={4}
  />
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "With key staff positions unfilled and no on-site first lady, the White House is trying to organize the most elaborate and heavily scrutinized public event of the year.",
        "image": "https://static01.nyt.com/images/2017/04/05/us/05eggroll2/05eggroll2-moth.jpg",
        "publisher": "New York Times",
        "title": "The Latest Test for the White House? Pulling Off Its Easter Egg Roll",
        "url": "http://www.nytimes.com/2017/04/11/us/politics/white-house-easter-egg-roll-trump.html",
      }
    }
    articleIndex={5}
  />
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "The White House is as much as two months behind recent standards for presidential transitions, leaving 90 percent of the positions considered critical to leadership unfilled. It did, however, manage to order the eggs for the Easter egg roll.",
        "image": "https://s15.postimg.org/6jredm53v/news.png",
        "publisher": "New York Times",
        "title": "Trump Administration Hunts for Easter Eggs, and Senior Staff",
        "url": "http://www.nytimes.com/video/us/politics/100000005018390/trump-administration-hunts-for-easter-eggs-and-senior-staff.html",
      }
    }
    articleIndex={6}
  />
  <Connect(ReaderListItem)
    article={
      Object {
        "description": "The revelations highlight the lax oversight at the A.T.F. that allowed agents and informants to spend millions while avoiding the normal accounting process.",
        "image": "https://static01.nyt.com/images/2017/04/11/us/11ATF/11ATF-moth.jpg",
        "publisher": "New York Times",
        "title": "Secret A.T.F. Account Paid for $21,000 Nascar Suite and Las Vegas Trip",
        "url": "http://www.nytimes.com/2017/04/11/us/politics/alcohol-tobacco-firearms-atf.html",
      }
    }
    articleIndex={7}
  />
</div>
`;
